cmd/gh/main_test.go

```diff
@@ -76,3 +88,128 @@ check your internet connection or https://githubstatus.com
 		})
 	}
 }
+
+func Test_IntegrationRepoFork(t *testing.T) {
+	// Set up temp dir
+	tempDir := t.TempDir()
+	oldWd, _ := os.Getwd()
+	assert.NoError(t, os.Chdir(tempDir))
+	t.Cleanup(func() { _ = os.Chdir(oldWd) })
+
+	// Set up server to handle network requests
+	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		path := r.URL.Path
+		gitPath := regexp.MustCompile(`\/someone\/repo\.git\/.*`)
+		forkPath := "/repos/owner/repo/forks"
+		switch {
+		case path == forkPath:
+			forkResult := fmt.Sprintf(`{
+				"node_id": "123",
+				"name": "repo",
+				"clone_url": "https://github.com/someone/repo.git",
+				"created_at": "%s",
+				"owner": {
+					"login": "someone"
+				}
+			}`, time.Now().Format(time.RFC3339))
+			_, _ = w.Write([]byte(forkResult))
+		case gitPath.MatchString(path):
+			w.WriteHeader(http.StatusOK)
+		default:
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}))
+	t.Cleanup(s.Close)
+	os.Setenv("HTTP_PROXY", s.URL)
+	t.Cleanup(func() { os.Unsetenv("HTTP_PROXY") })
+
+	// Set up config directory and config files
+	os.Setenv("GH_CONFIG_DIR", tempDir)
+	t.Cleanup(func() { os.Unsetenv("GH_CONFIG_DIR") })
+	hosts := heredoc.Doc(`
+    github.localhost:
+      user: monalisa
+      oauth_token: TOKEN
+      git_protocol: https
+	`)
+	err := ioutil.WriteFile("hosts.yml", []byte(hosts), 0771)
+	assert.NoError(t, err)
+
+	// Set up local git repo
+	gitExe, err := safeexec.LookPath("git")
+	assert.NoError(t, err)
+	cmd := exec.Command(gitExe, "init", "--quiet", "repo")
+	err = cmd.Run()
+	assert.NoError(t, err)
+	assert.NoError(t, os.Chdir("repo"))
+	url := fmt.Sprintf("%s/owner/repo.git", "http://github.localhost")
+	cmd2 := exec.Command(gitExe, "remote", "add", "origin", url)
+	err = cmd2.Run()
+	assert.NoError(t, err)
+	// Add remote as resolved to skip graphql resolution
+	cmd3 := exec.Command(gitExe, "config", "--add", "remote.origin.gh-resolved", "base")
+	err = cmd3.Run()
+	assert.NoError(t, err)
+	// Set up server as git proxy
+	cmd4 := exec.Command(gitExe, "config", "http.http://github.localhost.proxy", s.URL)
+	err = cmd4.Run()
+	assert.NoError(t, err)
+
+	// Skip checking for updates
+	os.Setenv("GH_NO_UPDATE_NOTIFIER", "1")
+	t.Cleanup(func() { os.Unsetenv("GH_NO_UPDATE_NOTIFIER") })
+
+	// Skip terminal colors
+	os.Setenv("NO_COLOR", "1")
+	t.Cleanup(func() { os.Unsetenv("NO_COLOR") })
+	os.Setenv("CLICOLOR", "0")
+	t.Cleanup(func() { os.Unsetenv("CLICOLOR") })
+
+	// Set integration test environment variable
+	os.Setenv("GH_INTEGRATION_TEST", "1")
+	t.Cleanup(func() { os.Unsetenv("GH_INTEGRATION_TEST") })
```

Comment:
Protip: there is already a `setenv` helper method in one of our tests that you could extract & reuse. Note that Go 1.17 has a setenv-like header for testing, but until we upgrade to it, we should keep using the helper.